### ISSUE TITLE
Gallery 2.0.0 Multi-upload error

### ISSUE THREAD

**Issue by gravadesign** [modxcms/Gallery#96]
Hi,
after updating to Gallery 2.0.0, from Gallery 1.7.1, on modx 2.8.4 the multi-upload function does not work, while the single upload works correctly.

The file is listed in the album with a broken image (not generate the thumbnail) (See screenshot down)
![Schermata 2022-07-13 alle 11 10 17](https://user-images.githubusercontent.com/5929863/178696868-1d7d2d44-7080-4d4f-9969-aa8fd602d8fa.png)

Why in the folder of this gallery was created the file at 0 byte, has not been uploaded (See screenshot down)
![Schermata 2022-07-13 alle 11 33 41](https://user-images.githubusercontent.com/5929863/178701820-1df4e046-e944-4800-a450-a96f93439af0.png)

Versions:
Modx: 2.8.4
PHP tested: 7.2, 7.3 and 7.4

This is the manager error:

```
13.07.2022	11:31:49	ERROR	/home/mydomain/public_html/core/components/gallery/model/gallery/galalbum.class.php	210
[Gallery] Could not create directory (possibly already exists?): assets/gallery/18/
13.07.2022	11:31:49	ERROR	/home/mydomain/public_html/core/components/gallery/processors/web/phpthumb.class.php	181
[phpThumbOf] Could not generate thumbnail: /home/mydomain/public_html/assets/gallery/18/2818.jpg - Debug: Array
(
    [0] => phpThumb() v1.7.16-202012161640 in file "phpthumb.class.php" on line 234
    [1] => ResolveFilenameToAbsolute() NOT prepending $this->config_document_root () to $filename (/assets/gallery/18/2818.jpg) resulting in ($AbsoluteFilename = "/assets/gallery/18/2818.jpg") in file "phpthumb.class.php" on line 1349
    [2] => resolvePath: /assets/gallery/18/2818.jpg (allowed_dirs: Array
(
)
) in file "phpthumb.class.php" on line 1183
    [3] => resolvePath: iteration, path=/assets/gallery/18/2818.jpg, base path = /home/mydomain/public_html/assets/components/gallery in file "phpthumb.class.php" on line 1202
    [4] => open_basedir: "/home/mydomain/:/opt/cpanel/ea-php72/root/usr:/usr/local/lib/php:/tmp:/etc/pki/tls/certs/ca-bundle.crt:/var/cpanel/php/sessions:/usr/local/bin/wp" in file "phpthumb.class.php" on line 1169
    [5] => resolvePath: stop at component 5 in file "phpthumb.class.php" on line 1218
    [6] => resolvePath: stop at path=/assets/gallery/18/2818.jpg in file "phpthumb.class.php" on line 1221
    [7] => resolvePath: no match, returning null in file "phpthumb.class.php" on line 1223
    [8] => !$this->config_allow_src_above_docroot therefore setting "/assets/gallery/18/2818.jpg" (outside "/home/mydomain/public_html/assets/components/gallery") to null in file "phpthumb.class.php" on line 1394
    [9] => setSourceFilename() set $this->sourceFilename to "" in file "phpthumb.class.php" on line 299
    [10] => setSourceFilename(/home/mydomain/public_html/assets/gallery/18/2818.jpg) set $this->sourceFilename to "/home/mydomain/public_html/assets/gallery/18/2818.jpg" in file "phpthumb.class.php" on line 299
    [11] => CleanUpCacheDirectory() set to purge (NULL days; NULL MB; NULL files) in file "phpthumb.class.php" on line 751
    [12] => CleanUpCacheDirectory() skipped because "/home/mydomain/public_html/assets/components/gallery/cache//phpThumbCacheStats.txt" is recently modified in file "phpthumb.class.php" on line 761
    [13] => $AvailableImageOutputFormats = array(text;ico;bmp;wbmp;gif;webp;png;jpeg) in file "phpthumb.class.php" on line 1012
    [14] => $this->thumbnailFormat set to $this->config_output_format "jpeg" in file "phpthumb.class.php" on line 1023
    [15] => $this->thumbnailFormat set to $this->f "jpeg" in file "phpthumb.class.php" on line 1030
    [16] => $this->thumbnailQuality set to "75" in file "phpthumb.class.php" on line 1040
    [17] => resolvePath: /home/mydomain/public_html/assets/gallery/18/2818.jpg (allowed_dirs: Array
(
)
) in file "phpthumb.class.php" on line 1183
    [18] => resolvePath: iteration, path=/home/mydomain/public_html/assets/gallery/18/2818.jpg, base path = /home/mydomain/public_html in file "phpthumb.class.php" on line 1202
    [19] => resolvePath: stop at component 8 in file "phpthumb.class.php" on line 1218
    [20] => resolvePath: stop at path=/home/mydomain/public_html/assets/gallery/18/2818.jpg in file "phpthumb.class.php" on line 1221
    [21] => resolvePath: path parsed, over in file "phpthumb.class.php" on line 1227
    [22] => $this->sourceFilename set to "/home/mydomain/public_html/assets/gallery/18/2818.jpg" in file "phpthumb.class.php" on line 933
    [23] => SetCacheFilename() _src set from md5($this->sourceFilename) "/home/mydomain/public_html/assets/gallery/18/2818.jpg" = "0d64bb2392cb1b45b624adb7059643bc" in file "phpthumb.class.php" on line 3616
    [24] => SetCacheFilename() _par set from md5(_farC_zc1_h80_w100_dpi150_q75) in file "phpthumb.class.php" on line 3655
    [25] => starting ExtractEXIFgetImageSize() in file "phpthumb.class.php" on line 3409
    [26] => getimagesize(/home/mydomain/public_html/assets/gallery/18/2818.jpg) failed in file "phpthumb.class.php" on line 3421
    [27] => getimagesize("/home/mydomain/public_html/assets/gallery/18/2818.jpg") failed in file "phpthumb.class.php" on line 3451
    [28] => $this->useRawIMoutput=true after checking $UnAllowedParameters in file "phpthumb.class.php" on line 1655
    [29] => phpThumb_tempnam() returning "/home/mydomain/php_sessions/pThumb7ZzIDm" in file "phpthumb.class.php" on line 4498
    [30] => ImageMagickSwitchAvailable(thumbnail) = 0 in file "phpthumb.class.php" on line 1581
    [31] => IMuseExplicitImageOutputDimensions = 0 in file "phpthumb.class.php" on line 1736
    [32] => getimagesize(/home/mydomain/public_html/assets/gallery/18/2818.jpg) FAILED with error "" in file "phpthumb.class.php" on line 1752
    [33] => ImageMagickSwitchAvailable(density) = 0 in file "phpthumb.class.php" on line 1581
    [34] => getimagesize(/home/mydomain/public_html/assets/gallery/18/2818.jpg) failed in file "phpthumb.class.php" on line 1932
    [35] => Remaining $this->fltr after ImageMagick: (array(0) { } ) in file "phpthumb.class.php" on line 2316
    [36] => ImageMagickSwitchAvailable(quality;interlace) = 0 in file "phpthumb.class.php" on line 1578
    [37] => ImageMagick called as ('' '/home/mydomain/public_html/assets/gallery/18/2818.jpg[0]' -resize '100x80' jpeg:'/home/mydomain/php_sessions/pThumb7ZzIDm' 2>&1) in file "phpthumb.class.php" on line 2334
    [38] => ImageMagick failed with message (sh: : command not found) in file "phpthumb.class.php" on line 2339
    [39] => deleting "/home/mydomain/php_sessions/pThumb7ZzIDm" in file "phpthumb.class.php" on line 2371
    [40] => ImageMagickThumbnailToGD() failed in file "phpthumb.class.php" on line 3458
    [41] => SetOrientationDependantWidthHeight() starting with ""x"" in file "phpthumb.class.php" on line 3388
    [42] => SetOrientationDependantWidthHeight() setting w="100", h="80" in file "phpthumb.class.php" on line 3404
    [43] => EXIF thumbnail extraction: (size=0; type=""; 0x0) in file "phpthumb.class.php" on line 3517
    [44] => starting SourceImageToGD() in file "phpthumb.class.php" on line 3781
    [45] => starting ImageCreateFromFilename(/home/mydomain/public_html/assets/gallery/18/2818.jpg) in file "phpthumb.class.php" on line 3696
    [46] => empty $filename or getimagesize(/home/mydomain/public_html/assets/gallery/18/2818.jpg) failed in file "phpthumb.class.php" on line 3745
    [47] => Populating $rawimagedata in file "phpthumb.class.php" on line 3753
    [48] => Not using EXIF thumbnail data because $this->exif_thumbnail_data is empty in file "phpthumb.class.php" on line 3853
    [49] => $this->gdimg_source is still empty in file "phpthumb.class.php" on line 3905
    [50] => ImageMagickThumbnailToGD() failed in file "phpthumb.class.php" on line 3907
    [51] => phpThumb() v1.7.16-202012161640
http://phpthumb.sourceforge.net

Error messages disabled.

edit phpThumb.config.php and (temporarily) set
$PHPTHUMB_CONFIG['disable_debug'] = false;
to view the details of this error in file "phpthumb.class.php" on line 4269
    [52] => SourceImageToGD() failed in file "phpthumb.class.php" on line 379
)

13.07.2022	11:31:49	ERROR	/home/mydomain/public_html/core/components/gallery/processors/web/phpthumb.class.php	187
PHP warning: filemtime(): stat failed for /home/mydomain/public_html/assets/components/gallery/cache/_home_mydomain_public_html_assets_gallery_18_2818.jpg.a5fc64dac5d325775e98e58ec08d6907.jpg
13.07.2022	11:31:49	ERROR	/home/mydomain/public_html/core/components/gallery/processors/web/phpthumb.class.php	199
PHP warning: file_get_contents(/home/mydomain/public_html/assets/components/gallery/cache/_home_mydomain_public_html_assets_gallery_18_2818.jpg.a5fc64dac5d325775e98e58ec08d6907.jpg): failed to open stream: No such file or directory
```

I have not find the solution, if anyone can help me I would be grateful.

**Comment by halftrainedharry** [comment]
For a solution, check this pull request #95.

**Comment by gravadesign** [comment]
Perfect, thank you very much!

**Comment by tiYusupov** [comment]
> For a solution, check this pull request #95.

just added this code, but still not working. Upload one photo or multi-upload. Still the same. How can I fix it? Thanks!

**Comment by halftrainedharry** [comment]
The only other issue with _Gallery 2.0.0_, that I know of, occurs when **MySQL 8** is used. This can be fixed with the PR #97.

If this isn't the case, then please create a new issue and add more information: Are there relevant error messages in the MODX error log? Are there errors in the browser's _developer tools_ (in the tab "Console" or in the responses of the AJAX-requests in the tab "Network")? What versions of MODX, PHP, MySQL are you using?

**Comment by tiYusupov** [comment]
> The only other issue with _Gallery 2.0.0_, that I know of, occurs when **MySQL 8** is used. This can be fixed with the PR #97.
> 
> If this isn't the case, then please create a new issue and add more information: Are there relevant error messages in the MODX error log? Are there errors in the browser's _developer tools_ (in the tab "Console" or in the responses of the AJAX-requests in the tab "Network")? What versions of MODX, PHP, MySQL are you using?

Thanks for reply! The problem was in Source Types. There were Gallery Albums. I changed it to File System and it worked.
